### PR TITLE
Configurable symbolic link following.

### PR DESCRIPTION
**Pull request by jacobstr**
Adds a followSymlink option to the FileFinder. Potentially useful when traversing
node_modules folders contain symbolic links - e.g. due to `npm link`. 

My personal use case is a bit messy to explain, see:

https://github.com/mover-io/jestsymlink

This might not be the best way to address the issue. It allows jest to traverse inside of a node_modules/symlinked_folder to populate the ResourceMap it while generating it's normalized id's.

### CODE REVIEW

**Comment by jacobstr** [comment]
This is a large api change unfortunately.  Adding another parameter to find and findNative would require 
I put effort into keeping this backwards compatible. The find / findNative methods are are exposed directly and those have been wrapped up to make them behave as before.

File: lib/FileFinder.js [hunk 1]
```diff
@@ -18,19 +18,74 @@ var fs = require('fs');
...
+ * @param  {bool}           options.useNative       Use os find instead of javascript's
+ * @param  {bool}           options.followSymlinks  Whether to follow symlinks.
+ * @param {Object} options
+ */
+function FileFinder(options) {
+  this.scanDirs = options && options.scanDirs || ['.'];
+  this.extensions = options && options.extensions || ['.js'];
+  this.ignore = options && options.ignore || null;
+  this.useNative = options && options.useNative || false;
+  this.followSymlinks = options && options.followSymlinks || true;
```
Comment (jacobstr) [inline lib/FileFinder.js]: This should default to false for backwards compatibility. Was testing this alongside jest.

**Comment by jacobstr** [comment]
Related: https://github.com/facebook/jest/issues/219

**Comment by facebook-github-bot-9** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Comment by cpojer** [comment]
We rewrote node-haste with a new implementation. Please feel free to reopen this pull request with a version against the new node-haste if you believe this issue still persists.

jest 0.8.0 will be launched with the new implementation of node-haste towards the end of the year. The integration work can be tracked in https://github.com/facebook/jest/pull/599

**Pull request closed without merging**